Foundation/URLCache.swift

```diff
@@ -243,11 +283,91 @@ open class URLCache : NSObject {
         usage of the on-disk cache. 
         @result the current usage of the on-disk cache of the receiver.
     */
-    open var currentDiskUsage: Int { NSUnimplemented() }
+    open var currentDiskUsage: Int {
+        return self.syncQ.sync { self.persistence?.cacheSizeInDesk ?? 0 }
+    }
+
 }
 
 extension URLCache {
     public func storeCachedResponse(_ cachedResponse: CachedURLResponse, for dataTask: URLSessionDataTask) { NSUnimplemented() }
     public func getCachedResponse(for dataTask: URLSessionDataTask, completionHandler: (CachedURLResponse?) -> Void) { NSUnimplemented() }
     public func removeCachedResponse(for dataTask: URLSessionDataTask) { NSUnimplemented() }
 }
+
+fileprivate struct _CachePersistence {
+
+    let path: String
+
+    var cacheSizeInDesk: Int {
+        do {
+            let subFiles = try FileManager.default.subpathsOfDirectory(atPath: path)
+            var total: Int = 0
+            for fileName in subFiles {
+                let attributes = try FileManager.default.attributesOfItem(atPath: path.appending(fileName))
+                total += (attributes[.size] as? Int) ?? 0
+            }
+            return total
```

Comment:
Is there plan to keep this value updated as cached responses are added and evicted? Feels like with a big cache and a slow disk, each of these calls might take a while.

Reply:
Let me add a `FIXME` comment, I will create a separate PR for optimizing this. My main focus in this PR is to create an initial setup and kick start the implementation. Hope that's ok.